src/sample/combinednr.jl

```diff
@@ -370,3 +370,11 @@ end
 function Base.length(nr::CombinedNextReaction)
     return length(nr.transition_entry)
 end
+
+function Base.haskey(nr::CombinedNextReaction{K,T}, clock) where {K,T}
+    if clock isa K
```

Comment:
You had an interesting thought that typeof(clock) might not be K. Would Julia handle this with two functions, one haskey(nr::CombinedNextReaction{K,T}, clock::K) and one with the generic clock? I'm not sure what's best. This is fine, though.